ledgerjs-fio/src/fio.ts

```diff
@@ -0,0 +1,335 @@
+    /**
+     * Get public key for the specified BIP 32 path.
+     *
+     * @param path The path. A path must begin with `44'/235'/0'/0/i`.
+     * @returns The public key (i.e. with chaincode).
+     *
+     * @example
+     * ```
+     * const publicKey = await fio.getPublicKey[[ HARDENED + 44, HARDENED + 235, HARDENED + 0, 0, 0 ]);
+     * console.log(publicKey);
+     * ```
+     */
+    async getPublicKey(
+        {path, show_or_not}: GetPublicKeyRequest
+    ): Promise<GetPublicKeyResponse> {
+        // validate the input
+        validate(isValidPath(path), InvalidDataReason.GET_EXT_PUB_KEY_PATHS_NOT_ARRAY)
+        const parsedPath = parseBIP32Path(path, InvalidDataReason.INVALID_PATH)
+
+        return interact(this._getPublicKey(parsedPath, show_or_not), this._send)
+    }
+
+    /** @ignore */
+    * _getPublicKey(path: ValidBIP32Path, show_or_not: boolean) {
+        const version = yield* getVersion()
+        return yield* getPublicKey(version, path, show_or_not)
+    }
+
+    /**
+     * Sign transaction.
+     *
+     * @param path The path. A path must begin with `44'/235'/0'/0/i`.
+     * @returns The public key (i.e. with chaincode).
+     *
+     * @example
+     * ```
+     * TODO
+     * const  = await fio.getPublicKey[[ HARDENED + 44, HARDENED + 235, HARDENED + 0, 0, 0 ]); TODO
+     * console.log(sign);
+     * ```
+     */
+    async signTransaction({path, chainId, tx}: SignTransactionRequest): Promise<SignTransactionResponse> {
+        validate(isHexString(chainId), InvalidDataReason.INVALID_CHAIN_ID)
+
+        const parsedPath = parseBIP32Path(path, InvalidDataReason.INVALID_PATH)
+        const parsedTx = parseTransaction(chainId, tx)
+
+        return interact(this._signTransaction(parsedPath, chainId as HexString, parsedTx), this._send)
+    }
+
+    /** @ignore */
+    * _signTransaction(parsedPath: ValidBIP32Path, chainId: HexString, tx: ParsedTransaction) {
+        const version = yield* getVersion()
+        return yield* signTransaction(version, parsedPath, chainId, tx)
+    }
+
+    /**
+     * Runs unit tests on the device (DEVEL app build only)
+     */
+    async runTests(): Promise<void> {
+        return interact(this._runTests(), this._send)
+    }
+
+    /** @ignore */
+    * _runTests(): Interaction<void> {
+        const version = yield* getVersion()
+        return yield* runTests(version)
+    }
+
+}
+
+/**
+ * Get FIO app version [[Fio.getVersion]] response data
+ * @category Main
+ */
+export type GetVersionResponse = {
+    version: Version
+    compatibility: DeviceCompatibility
+}
+
+/**
+ * Get device serial number ([[Fio.getSerial]]) response data
+ * @category Main
+ */
+export type GetSerialResponse = Serial
+
+/**
+ * Get public key ([[Fio.getPublicKey]]) request data
+ * @category Main
+ * @see [[GetPublicKeyResponse]]
+ */
+export type GetPublicKeyRequest = {
+    /** Paths to public keys which should be derived by the device */
+    show_or_not: boolean, 
+    path: BIP32Path
+}
```

Comment:
Incorrect information. Currently the jsdoc applies to the `show_or_not` parameter.
```suggestion
export type GetPublicKeyRequest = {
    /** Path to public key which should be derived by the device. A path must begin with `44'/235'/0'/0/i`. */
    path: BIP32Path,
    /** Please fill this description as well */
    show_or_not: boolean, 
}
```